core/src/main/java/com/amplifyframework/analytics/AnalyticsCategory.java

```diff
@@ -60,21 +60,25 @@ public CategoryType getCategoryType() {
     }
 
     @Override
-    public void identifyUser(@NonNull String userId, @NonNull AnalyticsProfile profile) {
-        throw new UnsupportedOperationException("This operation is currently not supported.");
+    public void identifyUser(@NonNull String userId, @NonNull AnalyticsUserProfile profile) {
+        if (enabled) {
+            getSelectedPlugin().identifyUser(userId, profile);
+        }
     }
 
     @Override
     public void disable() {
         synchronized (LOCK) {
             enabled = false;
+            getSelectedPlugin().disable();
         }
     }
 
     @Override
     public void enable() {
         synchronized (LOCK) {
```

Comment:
Which components is this lock guarding? Is it to synchronize access to the boolean? Or what's the intention?

Reply:
Its guarding the access to boolean flag `enabled`.